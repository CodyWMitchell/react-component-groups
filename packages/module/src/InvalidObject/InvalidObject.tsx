import { Button, EmptyState, EmptyStateBody, EmptyStateFooter, EmptyStateHeader, EmptyStateVariant } from '@patternfly/react-core';

import NotFoundIcon from '../NotFoundIcon/NotFoundIcon';
import React from 'react';
import clsx from 'clsx';
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  "invalidObject": {
    display: "flex",
    justifyContent: "center",
    flexDirection: "column",
    alignItems: "center",
    textAlign: "center",
    h1: { marginBottom: "30px" },
    svg: { marginBottom: "30px" },
    button: { fontSize: "20px" }
  },
  "invalidObjectSorryText": { maxWidth: "70%" },
});

export interface InvalidObjectProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement> {
  /** Custom landing page button URL */
  toLandingPageUrl?: string;
  /** Custom return to landing page text */
  toLandingPageText?: React.ReactNode;
}


const InvalidObject: React.FunctionComponent<InvalidObjectProps> = ({
  toLandingPageUrl = `${window.location.origin}`,
  toLandingPageText = 'Return to homepage'
}: InvalidObjectProps) => {
  const classes = useStyles();
  const invalidObjectClasses = clsx("pf-v5-l-page__main-section", "pf-v5-c-page__main-section", classes.invalidObject)
  return (
    <EmptyState variant={EmptyStateVariant.full} className={invalidObjectClasses}>
      <EmptyStateHeader titleText='We lost that page' icon={<NotFoundIcon />} headingLevel='h1' />
      <EmptyStateBody>Let&apos;s find you a new one. Try a new search or return home.</EmptyStateBody>
      <EmptyStateFooter>
        <Button variant="link" component="a" href={toLandingPageUrl}>
          {toLandingPageText}
        </Button>
      </EmptyStateFooter>
    </EmptyState>
  );
};

export default InvalidObject;