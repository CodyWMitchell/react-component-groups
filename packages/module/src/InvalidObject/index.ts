export { default } from './InvalidObject';
export { default as InvalidObject } from './InvalidObject';
export { default as Icon404 } from '../NotFoundIcon/NotFoundIcon';