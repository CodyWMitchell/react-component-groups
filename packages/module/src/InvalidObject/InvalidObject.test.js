import InvalidObject from './InvalidObject';
import React from 'react';
import { render } from '@testing-library/react';
import toJson from 'enzyme-to-json';

describe('InvalidObject component', () => {
  test('should render', () => {
    const wrapper = render(<InvalidObject />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});