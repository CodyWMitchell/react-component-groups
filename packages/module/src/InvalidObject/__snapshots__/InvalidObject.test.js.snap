// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidObject component should render 1`] = `undefined`;
